quantization: fix scale+zp serialization of quantized BatchNorm{2|3}d

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70432

Summary:

Scale and zero_point need to be buffers for serialization to work
on them properly.  This PR moves them to buffers.  This is BC breaking,
but the "before" state was completely broken (scale + zp were not
serialized at all) so there is no value in trying to handle it.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_batch_norm2d_serialization
python test/test_quantization.py TestStaticQuantizedModule.test_batch_norm3d_serialization
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33330022](https://our.internmc.facebook.com/intern/diff/D33330022)

vkuzo committed:
[wip] fix scale+zp serialization of quantized BatchNorm{2|3}d

Summary:

Not for review yet.

Scale and zero_point need to be buffers for serialization to work
on them properly.  For now, this PR fixes the issue without dealing
with BC, just so we can run CI and see how many callsites expect
the old behavior.  CI results will determine next steps.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_batch_norm2d_serialization
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33330022).

vkuzo committed:
Update on "[wip] fix scale+zp serialization of quantized BatchNorm{2|3}d"


Summary:

Not for review yet.

Scale and zero_point need to be buffers for serialization to work
on them properly.  For now, this PR fixes the issue without dealing
with BC, just so we can run CI and see how many callsites expect
the old behavior.  CI results will determine next steps.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_batch_norm2d_serialization
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33330022](https://our.internmc.facebook.com/intern/diff/D33330022)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33330022).

vkuzo committed:
Update on "quantization: fix scale+zp serialization of quantized BatchNorm{2|3}d"


Summary:

Scale and zero_point need to be buffers for serialization to work
on them properly.  This PR moves them to buffers.  This is BC breaking,
but the "before" state was completely broken (scale + zp were not
serialized at all) so there is no value in trying to handle it.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_batch_norm2d_serialization
python test/test_quantization.py TestStaticQuantizedModule.test_batch_norm3d_serialization
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33330022](https://our.internmc.facebook.com/intern/diff/D33330022)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33330022).

jerryzh168 commented on test/quantization/core/test_quantized_module.py:
```diff
@@ -644,6 +644,50 @@ def test_batch_norm3d(self):
 
+    def _test_batch_norm_serialization(self, get_model, data1, data2):
+        m1 = get_model()
+        m1.qconfig = torch.quantization.default_qconfig
```
nit: torch.ao.quantization

jerryzh168 reviewed (approved):
Looks good, thanks!

vkuzo committed:
Update on "quantization: fix scale+zp serialization of quantized BatchNorm{2|3}d"


Summary:

Scale and zero_point need to be buffers for serialization to work
on them properly.  This PR moves them to buffers.  This is BC breaking,
but the "before" state was completely broken (scale + zp were not
serialized at all) so there is no value in trying to handle it.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule.test_batch_norm2d_serialization
python test/test_quantization.py TestStaticQuantizedModule.test_batch_norm3d_serialization
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33330022](https://our.internmc.facebook.com/intern/diff/D33330022)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33330022).
